models/event.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const Build = require('./build');
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+const Pipeline = require('./pipeline');
+const Scm = require('../core/scm');
+const Workflow = require('../config/workflow');
+
+const buildId = Joi.reach(Build.base, 'id');
+
+const MODEL = {
+    id: Joi
+        .string().hex().length(40)
+        .description('Identifier of this build')
+        .example('4b8d9b530d2e5e297b4f470d5b0a6e1310d29c5e'),
+    builds: Joi
```

Comment:
> I worry about a possible race condition

The way we were thinking about it was that you can determine which builds are attached to this event when you access it. But, like I said, this is an optimization. You'll still need to query the related builds for the full data set, and we should treat the build table as the source of truth if it comes down to a race condition. 

One of many philosophies with NoSQL usage is that you promote some small amount of redundancy across tables. For instance, the Github API carries some redundancy with its calls by providing some portion of User information in their commit data. 

I'm not saying we should follow suit; it's just difficult to break out of that mindset when your initial datastore is DynamoDB. We have to support SQL-based implementations too, so we can't carry that type of philosophy if we need to support normalized SQL tables.